enable hl_thread_id for OpenBSD using getthrid(2)
pthread_setname_np -> pthread_set_name_np

Index: src/std/thread.c
--- src/std/thread.c.orig
+++ src/std/thread.c
@@ -26,6 +26,11 @@
 typedef struct _hl_semaphore hl_semaphore;
 typedef struct _hl_condition hl_condition;
 
+#ifdef HL_OPENBSD
+#include <pthread_np.h>
+#define pthread_setname_np(a,b) pthread_set_name_np(a,b)
+#endif
+
 #if !defined(HL_THREADS)
 
 struct _hl_mutex {
@@ -838,6 +843,8 @@ HL_PRIM int hl_thread_id() {
 	uint64_t tid64;
 	pthread_threadid_np(NULL, &tid64);
 	return (pid_t)tid64;
+#elif defined(HL_OPENBSD)
+	return getthrid();
 #elif defined(SYS_gettid) && !defined(HL_TVOS)
 	return syscall(SYS_gettid);
 #else
